### PR TITLE
Add description for void key and parallel source

### PR DESCRIPTION
**Pull request by mrshenli**

### CODE REVIEW

File: docs/beamrunner/beamrunner-X-performance.md [hunk 1]
```diff
@@ -59,6 +59,43 @@ option `--parallelWidths=4,Serial=1,VeryParallel=8` would disable
 parallelism for `Serial` while using higher parallelism for `VeryParallel`
 than the rest of the application.
 
+`PCollection`s of [`KV`](https://github.com/apache/beam/blob/release-2.4.0/sdks/java/core/src/main/java/org/apache/beam/sdk/values/KV.java) elements are partitioned based on keys. The same key always goes into
```
Comment (pgerv12) [inline docs/beamrunner/beamrunner-X-performance.md]: "The same key will always go into"

File: docs/beamrunner/beamrunner-X-performance.md [hunk 2]
```diff
@@ -59,6 +59,43 @@ option `--parallelWidths=4,Serial=1,VeryParallel=8` would disable
...
+scheme. It is applications' responsibility to balance workloads across keys.
+Skewed workloads can overwhelm hot keys and hamper performance. Since the
+value of keys are determined by applications, the runner cannot easily
+detect or predict keys. If a `PCollection` contains fewer keys than the
+parallel width, some parallel channels will stay idle, wasting system
+resources. One special case is a `PCollection` using `Void` key. In this
+situation, the runner can identify that there is only one global key and
+run the transform in a non-parallelized form. The following code snippet
+demonstrates an example. As the `AddKey` transform add the same `Void` key
+to all elements, the downstream `GBK` cannot run in parallel. If the
```
Comment (pgerv12) [inline docs/beamrunner/beamrunner-X-performance.md]: After "cannot run in parallel".  Add the following: 
```
If a user specifies a parallel width (greater than 1) for the entire pipeline,
the runner may silently ignore the parallel width for the section of the pipeline 
that uses the `Void` keys. If a user explicitly specifies a parallel width for the `GBK` in the example, the runner, again, may ignore the specified width.
```
Don't know if we need to say a warning message will be printed.

**Comment by ghost** [comment]
Looks good! Two minor suggestions:
1. I'd change "please expect the runner to print a warning message and ignores" to "the runner will print a warning message and ignore", or at least "please expect the runner to print a warning message and ignore".
2. I had a link for `DoFn` to the API docs page, rather than the source code. I have only a slight preference for that, but either way we should be consistent, so if we stick with source that one should get updated.

**Comment by mrshenli** [comment]
Thanks for the comments! I will address. For the link, I think the API docs page is better.

**Review by pgerv12 — APPROVED** [review]

**Pull request merged**